### PR TITLE
Pivotal ID: 175632448: Extra persistence layer interfaces to avoid dependencies on SQL classes

### PR DESCRIPTION
**Pull request by Juan-EBI**
https://www.pivotaltracker.com/n/projects/1189310

### CODE REVIEW

File: submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/service/PersistenceService.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package ac.uk.ebi.biostd.persistence.common.service
+
+@Suppress("TooManyFunctions")
+interface PersistenceService {
```
Comment (jhoanmanuelms) [inline submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/service/PersistenceService.kt:4]: Just 5 triggers too many functions??

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/ext/getSimpleByAccNo.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package ac.uk.ebi.biostd.submission.ext
+
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionQueryService
+import ebi.ac.uk.extended.mapping.to.toSimpleSubmission
+
+fun SubmissionQueryService.getSimpleByAccNo(accNo: String) = getExtByAccNo(accNo).toSimpleSubmission()
```
Comment (jhoanmanuelms) [inline submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/ext/getSimpleByAccNo.kt:6]: Why is this an extension?

File: submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/service/SubmissionOperations.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import ac.uk.ebi.biostd.persistence.common.request.SaveSubmissionRequest
+import ac.uk.ebi.biostd.persistence.common.request.SubmissionFilter
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.extended.model.FileMode
+import org.springframework.data.domain.Page
+import java.io.File
+import java.time.OffsetDateTime
+
+interface SubmissionRequestService {
+    fun saveAndProcessSubmissionRequest(saveRequest: SaveSubmissionRequest): ExtSubmission
```
Comment (jhoanmanuelms) [inline submission/persistence-common-api/src/main/kotlin/ac/uk/ebi/biostd/persistence/common/service/SubmissionOperations.kt:13]: As discussed in my last PR this should be separated?

**Review by jhoanmanuelms — APPROVED** [review]
LGTM. Please consider my comments before merging.

File: submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/ext/getSimpleByAccNo.kt (hunk 2 above)
Comment (Juan-EBI) [reply submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/ext/getSimpleByAccNo.kt:6]: This service comes from the persistence module so if we won't allow it in class we need to add the serialization module as a dependency on persistence. As that is not necessary at all. I Declared this an extension of main product (that already depends on the serialization model)

**Pull request merged**